module5/projects/MVCLevelUp.md

```diff
@@ -0,0 +1,101 @@
+* Utilize the skills for building a high quality .NET applications in a more open ended environment
+* Build familiarity with the maintenance phase of the SDLC
+* Handle navigating a new team and codebase
+
+## Output Goal
+* An even better project to add to your professional portfolio!
+
+## Overview
+
+Over the past week you've all learned many tools to improve the quality of your applications and level up your .NET skills!
+
+These tools include:
+* Error Handling and Validation
+* Logging
+* Refactoring
+* Writing Good Documentation
+
+This week you'll have the opportunity to return to your MVC group project from Mod 4 and level up your application by working on a sprint fully focused on "Maintenance"!
+
+What is maintenance, with regards to the Software Development Life Cycle (SDLC)?
+
+![SDLC](/assets/images/module5/week2/SDLC.png)
+
+I really like this explanation from [this](https://www.geeksforgeeks.org/software-engineering-software-maintenance/) post on Geeks for Geeks:
+
+"Software Maintenance refers to the process of modifying and updating a software system after it has been delivered to the customer. This can include fixing bugs, adding new features, improving performance, or updating the software to work with new hardware or software systems. The goal of software maintenance is to keep the software system working correctly, efficiently, and securely, and to ensure that it continues to meet the needs of the users.
+
+Software maintenance is a continuous process that occurs throughout the entire life cycle of the software system."
+
+On your future teams, you will probably not spend an entire sprint only on maintenance. Your sprint will probably be a combination of new features and maintenance. [Here](https://www.agilecoachjournal.com/2018-12-18/maintenance-patterns-for-scrum-teams) is further explanation on how that might look like.
+
+![MaintenanceNewFeatures](/assets/images/module5/week2/MaintenanceNewFeatures.png)
+
+But in a class setting, we mainly focus on new features, so for this project we're going all in on maintenance. You will be adding **no new features** only leveling up what's already there. 
+
+## Project Requirements
+
+Each team is responsible for implementing each of the following topics at least once:
+
+* Error Handling and Validation
+* Logging
+* Refactoring
+* Documentation Improvement
+
+You can also work on tasks related to:
+* Fixing bugs
+* Making small improvements to the user experience
+
+Each team must follow a git workflow and create pull requests instead of coding directly on the main branch. Each change must go through code review before merging unless it was build entirely with pair programming.
+
+### Working on a New Team and Codebase
+
+For days 2, 3 and 4 of the project one team member will swap to a new team for the day. Joining a new team and codebase is HARD, but it's something you will all need to do in your first job and a great learning experience to practice with this skill now.

```

Comment:
I wonder if we could re-word this so that it is clear that it is difficult for both the person joining **and** the team that is getting a new member.  That way if there are one or two students who don't get the opportunity to swap, they can still see the benefit.  (I'm thinking more for future, larger cohorts)

Reply:
I think it's a good callout that it can be difficult for both the existing and new team members.

Although in future larger cohorts if we have maybe a group of 5 I could see two folks swapping at once (like we will already have to do for our group of 4).